### PR TITLE
Added in support for RangeExec

### LINKED ISSUE #385: [FEA] Support RangeExec on the GPU
`spark.range` can be helpful in many queries, supporting it on the GPU would be great and we should be able to do it with existing cudf functionality.

### PR DESCRIPTION
**Pull request by revans2**
This fixes #385

### CODE REVIEW

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala [hunk 1]
```diff
@@ -170,11 +170,133 @@ case class GpuFilterExec(condition: Expression, child: SparkPlan)
...
+                  number < safePartitionEnd
+                } else {
+                  number > safePartitionEnd
+                }
+              } else false
+
+            override def next(): ColumnarBatch =
+              withResource(new NvtxWithMetrics("GpuRange", NvtxColor.DARK_GREEN, totalTime)){
+                _ =>
+                  GpuSemaphore.acquireIfNecessary(TaskContext.get())
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala:254]: Use `taskContext` here to avoid the extra thread-local lookup?

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Pull request merged**